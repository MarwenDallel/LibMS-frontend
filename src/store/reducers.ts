/**
 * Combine all reducers in this file and export the combined reducers.
 */
import { combineReducers } from '@reduxjs/toolkit';
import userGreetingReducer from 'app/pages/UserGreeting/slice';
import { InjectedReducersType } from 'utils/types/injector-typings';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(
  injectedReducers: InjectedReducersType = { userProfile: userGreetingReducer },
) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  if (Object.keys(injectedReducers).length === 0) {
    return state => state;
  } else {
    return combineReducers({
      ...injectedReducers,
    });
  }
}
