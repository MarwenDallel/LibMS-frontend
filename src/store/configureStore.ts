/**
 * Create the store with dynamic reducers
 */

import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { createInjectorsEnhancer } from 'redux-injectors';
import createSagaMiddleware from 'redux-saga';
import createReduxWaitForMiddleware from 'redux-wait-for-action';
import { createReducer } from './reducers';

export function configureAppStore() {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const reduxWaitForMiddleware = createReduxWaitForMiddleware();
  const { run: runSaga } = sagaMiddleware;

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware, reduxWaitForMiddleware];

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ];

  const store = configureStore({
    reducer: createReducer(),
    middleware: [
      ...getDefaultMiddleware({
        serializableCheck: {
          // Ignore these field paths in all actions
          ignoredActionPaths: ['payload.file', 'payload.image'],
        },
      }),
      ...middlewares,
    ],
    devTools: process.env.NODE_ENV !== 'production',
    enhancers,
  });

  return store;
}
