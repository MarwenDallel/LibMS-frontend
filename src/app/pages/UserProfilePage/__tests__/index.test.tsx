import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import * as React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { configureAppStore } from 'store/configureStore';
import { UserProfilePage } from '..';

const store = configureAppStore();

const renderWithRouter = (component: JSX.Element) => {
  const history = createMemoryHistory();
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>{' '}
      </Provider>,
    ),
  };
};

describe('<UserProfilePage  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = renderWithRouter(<UserProfilePage />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
