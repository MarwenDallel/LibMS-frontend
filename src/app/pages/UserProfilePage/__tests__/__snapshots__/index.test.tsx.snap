// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfilePage  /> should match snapshot 1`] = `
.c2 {
  background-color: #203549;
}

.c1 {
  background-color: #007ea4;
}

.c0 {
  background-color: #203549;
}

<div
  class="d-flex flex-column min-vh-100"
>
  <div
    class="Header__Div-sc-72bdxd-0"
  >
    <nav
      class="c0 navbar navbar-expand navbar-light"
    >
      <div
        class="container"
      >
        <a
          href="/"
        >
          <img
            alt=""
            height="122"
            src="SMU-LOGO.png"
            width="263"
          />
        </a>
      </div>
    </nav>
    <nav
      class="c1 navbar navbar-expand-md navbar-dark navbar navbar-expand-lg navbar-light"
    >
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="justify-content-center justify-content-md-between container"
        >
          <div
            class="font-weight-light mr-0 text-white"
          >
            My SMU-Library Account
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    class="text-left wrapper flex-grow-1 container"
  >
    <div
      class="d-block mt-5 mb-5 font-weight-light"
      style="font-size: 2em;"
    >
      Welcome
       
      <div
        class="spinner-grow"
      />
    </div>
  </div>
  <footer
    class="c2 text-lg-start"
  >
    <div
      class="container"
    >
      <img
        alt=""
        src="FOOTER-LOGO.png"
      />
    </div>
  </footer>
</div>
`;
